File: solutions/java/space-age/1/src/main/java/SpaceAge.java
class SpaceAge {
    double ageInSecs;

    SpaceAge(double seconds) {
        ageInSecs = seconds;
    }

    double getSeconds() {
        return(ageInSecs);
    }

    double onEarth() {
        return(ageInSecs/31557600);
    }

    double onMercury() {
        return((ageInSecs/31557600)/0.2408467);
    }

    double onVenus() {
        return((ageInSecs/31557600)/0.61519726);
    }

    double onMars() {
        return((ageInSecs/31557600)/1.8808158);
    }

    double onJupiter() {
        return((ageInSecs/31557600)/11.862615);
    }

    double onSaturn() {
        return((ageInSecs/31557600)/29.447498);
    }

    double onUranus() {
        return((ageInSecs/31557600)/84.01684);
    }

    double onNeptune() {
        return((ageInSecs/31557600)/164.79132 );
    }

}

File: solutions/csharp/space-age/1/SpaceAge.cs
using System;

public class SpaceAge
{
    long secs;
    public SpaceAge(long seconds)
    {

        secs = seconds;
  
    }

    public double OnEarth()
    {
        return secs / 31557600.0;
    }

    public double OnMercury()
    {
        return OnEarth() / 0.2408467;
    }

    public double OnVenus()
    {
        return OnEarth() / 0.61519726;
    }

    public double OnMars()
    {
        return OnEarth() / 1.8808158;
    }

    public double OnJupiter()
    {
        return OnEarth() / 11.862615;
    }

    public double OnSaturn()
    {
        return OnEarth() / 29.447498;
    }

    public double OnUranus()
    {
        return OnEarth() / 84.016846;
    }

    public double OnNeptune()
    {
        return OnEarth() / 164.79132;
    }
}